enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,893 @@
+(container images and metadata) that are required to perform the upgrade plus
+the images required for the custom workloads, and writes it to a USB stick (or
+any other suitable media) that will be handed over to the technicians
+responsible for performing the upgrades, together with documentation explaining
+how to use it.
+
+1. The technicians receive copies of the USB stick and the corresponding
+documentation.
+
+1. The technician goes to the cluster site and uses the upgrade bundle inside
+the USB stick to perform the upgrade. The documentation included in the bundle
+will basically ask the technician to plug the USB stick in one of the nodes of
+the cluster and then provide simple instructions to verify that the upgraded has
+been applied correctly. This step is potentially repeated multiple times by the
+same technician for multiple clusters using the same USB stick or copies of it.
+
+Note that the upgrade bundle should not be specific for a particular cluster,
+only for the OpenShift architecture and version. Technicians should be able to
+use that package for any cluster with that architecture and version.
+
+### API Extensions
+
+There are no new object kinds introduced by this enhancement, but new fields
+will be added to existing `ClusterVersion` and `ContainerRuntimeConfig` objects.
+Those fields are described in detail in the implementation details section below,
+this is a summary of those new `CluserVersion` fields:
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: ClusterVersion
+metadata:
+  name: version
+spec:
+  desiredUpdate:
+
+    # Indicates if the upgrade should be performed immediately or if it should
+    # be held for a later time. The default is `false` which means that the
+    # upgrade will be performed immediately (once the images are pre-loaded, if
+    # so requested).
+    hold: true|false
+
+    bundle:
+      # Indicates if use of a bundle is enabled. The default is `true`. If
+      # set to `false` then the `desiredUpdate.image` or `desiredUpdate.version`
+      # field will need to be populated. The cluster version operator will then
+      # pre-load the release images from the registry servers configured in the
+      # cluster and will not use or require a bundle.
+      enabled: true|false
+
+      # Indicates if the cluster version operator should monitor devices
+      # to automatically detect when an upgrade bundle is available. The
+      # defaut is `false`.
+      monitor: true|false
+
+      # Indicates the location of the upgrade bundle when `monitor` is `false`.
+      file: /root/m-bundle.tar # Can be a file.
+      file: /dev/sdb # Or a device.
+
+      # This is an optional digest of the bundle. If set then the cluster version
+      # operator will ensure that the bundle is used only if it matches this
+      # digest. This is intended to prevent use of a wrong USB stick, or use of
+      # a perfectly fine USB stick in the wrong cluster.
+      digest: sha256:....
```

Comment:
How is this value updated before a bundle arrives at the cluster to be installed?

Reply:
The administrator of the cluster sets it in advance, before the technician arrives in the cluster site and plugs the USB stick.

Reply:
If this is a completely disconnected environment, with no remote access at all, then this is impossible. The `digest` field would need to be empty, or else the technician should be provided with instructions to do that change herself.